src/app/orders/service/orders.service.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class OrdersService {
+  apiUrl = 'http://localhost:3000/api';
+
+  constructor(public http: HttpClient) {
+  }
+
+  getOrders() {
+    return new Promise(resolve => {
+      this.http.get(this.apiUrl + '/orders').subscribe(data => {
+        resolve(data);
+      }, err => {
+        console.log(err);
```

Comment:
For rejected situaction, you can save somewehere information about error and show simple information at order tab. 

You can create variable at here (OrdersService)  for failure and use it in order component to get info for user about problems.
Or you can use here resolve() with feedback to component with failure and then ...